apps/web/src/components/layout/components/v2/SyncInfoModalTrigger.tsx

```diff
@@ -15,11 +52,65 @@ export function SyncInfoModalTrigger() {
 
   return (
     <>
-      <Button size="xs" Icon={IconOutlineCloudUpload} onClick={toggleSyncInfoModalShow}>
-        Sync
-      </Button>
+      <Tooltip
+        disabled={isSynced}
+        width="auto"
+        label={<Text>You have un-synced changes in your local application.</Text>}
+        withinPortal
+      >
+        <Container>
+          <Button size="xs" Icon={IconOutlineCloudUpload} onClick={toggleSyncInfoModalShow}>
+            Sync
+          </Button>
+          <When truthy={!isSynced}>
+            <Dot />
+          </When>
+        </Container>
+      </Tooltip>
+
       {/** TODO: use a modal manager */}
-      <SyncInfoModal isOpen={showSyncInfoModal} toggleOpen={toggleSyncInfoModalShow} />
+      <SyncInfoModal isOpen={showSyncInfoModal} toggleOpen={toggleSyncInfoModalShow} refetchOriginWorkflows={refetch} />
     </>
   );
 }
+
+function createHash(workflowDefine) {
+  return CryptoJS.SHA256(JSON.stringify(workflowDefine || '')).toString(CryptoJS.enc.Hex);
+}
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
☑ **chore:** Please extract out of this file and add types‏

Reply:
done